src/steamship/agents/memory/chathistory.py

```diff
@@ -47,20 +80,54 @@ def get_or_create(
 
         if file is None:
             tags = tags or []
+            index_handle = str(uuid.uuid4())
             tags.append(Tag(kind=TagKind.DOCUMENT, name=DocTag.CHAT))
             tags.append(Tag(kind=TagKind.CHAT, name=ChatTag.CONTEXT_KEYS, value=context_keys))
+            tags.append(
+                Tag(
+                    kind=TagKind.CHAT,
+                    name=ChatTag.INDEX_HANDLE,
+                    value={TagValueKey.STRING_VALUE: index_handle},
+                )
+            )
 
             blocks = []
             file = File.create(
                 client=client,
                 blocks=blocks,
                 tags=tags,
             )
-        return ChatHistory(file)
+        else:
+            index_handle = ChatHistory._get_index_handle_from_file(file)
+
+        embedding_index = ChatHistory._get_embedding_index(client, index_handle)
+
+        return ChatHistory(file, embedding_index)
+
+    def _chunk_text(self, text: str) -> List[Tag]:
```

Comment:
Thinking out loud:

You know... the VectorSearchLearnerTool [1] actually does similar things... Would we want to leverage that as a reusable code asset in addition to an Agent-usable Tool? That might be an interesting way to consolidate the implementation. (At the cost of potentially trying to force two things to be the same when they might want to be 10% different)



[1] https://github.com/steamship-core/python-client/blob/main/src/steamship/agents/tools/question_answering/vector_search_learner_tool.py